net/tsdial/tsdial.go

```diff
@@ -91,10 +113,51 @@ func (d *Dialer) SetExitDNSDoH(doh string) {
 	}
 }
 
+func (d *Dialer) Close() {
+	d.mu.Lock()
```

Comment:
looks like a bug to not Unlock this too. big comment if that's your intention.

but even though that's clever, it usually leads to pain. e.g. it prevents double Closes which should usually be allowed.

so I'd add an Unlock and add a `closed bool`

Reply:
Done